project_euler/problem_63/sol1.py

```diff
@@ -0,0 +1,28 @@
+"""
+The 5-digit number, 16807=75, is also a fifth power. Similarly, the 9-digit number,
+134217728=89, is a ninth power.
+How many n-digit positive integers exist which are also an nth power?
+"""
+
+"""
+The maximum base can be 9 because all n-digit numbers < 10^n.
+Now 9**23 has 22 digits so the maximum power can be 22.
+Using these conclusions, we will calculate the result.
+"""
+
+
+def compute_nums(max_base: int = 10, max_power: int = 22) -> int:
+    """
+    Returns the count of all n-digit numbers which are nth power
+    >>> compute_nums(10, 22)
+    49
+    """
+    bases = list(range(1, max_base))

```

Comment:
If max_base and/or max_power is large then the list() call could occupy a lot of RAM and be slower than using the range iterator without the list() call.

Reply:
```
import sys
>>> sys.getsizeof(range(1000000))
48
>>> sys.getsizeof(list(range(1000000)))
9000112
```

Reply:
Should I consider removing the parameters?

Reply:
Ok. I got it.